### PR TITLE
allow tensor in DiagonalNormal dimension

### PR DESCRIPTION
**Pull request by arrjon**
This pull request updates the `DiagonalNormal` distribution implementation to support multi-dimensional (non-vector) distributions instead of being limited to vectors. The main changes involve replacing the single `dim` attribute with a tuple `dims`, and updating all relevant logic to handle arbitrary shapes.

This is necessary if we pass not a vector but something else to inference networks like diffusion, flow matching etc.

### CODE REVIEW

**Review by stefanradev93 — COMMENTED** [review]
Hi Jonas, thanks for the PR. See individual comments.

File: bayesflow/approximators/continuous_approximator.py
Comment (stefanradev93) [inline bayesflow/approximators/continuous_approximator.py:1]: This eliminates the possibility to sample for multiple batch dimensions, for instance, `(batch_size, time, num_samples,...)` where the target is still a vector.

File: bayesflow/distributions/diagonal_normal.py
Comment (stefanradev93) [inline bayesflow/distributions/diagonal_normal.py:1]: I would use `ops.sum`.

File: bayesflow/approximators/continuous_approximator.py
Comment (stefanradev93) [reply bayesflow/approximators/continuous_approximator.py:1]: I guess to differentiate the cases, we need to fetch the tensor dim of the target which is known by the latent distro.

File: bayesflow/distributions/diagonal_normal.py
Comment (arrjon) [reply bayesflow/distributions/diagonal_normal.py:1]: I did that before and it raised an error with jax as far as I can remember

File: bayesflow/approximators/continuous_approximator.py
Comment (arrjon) [reply bayesflow/approximators/continuous_approximator.py:1]: True, I will change this

File: bayesflow/distributions/diagonal_normal.py
Comment (stefanradev93) [reply bayesflow/distributions/diagonal_normal.py:1]: Oh, never mind, it's a constant so `sum` is good enough.

**Comment by arrjon** [comment]
I changed the batch shape now to
```
target_dim = self.inference_network.base_distribution.dims
batch_shape = keras.ops.shape(inference_conditions)[: -len(target_dim)]
```
that should cover now more use cases as the shape is inferred from the conditions, but the base shape is also considered.
